### ISSUE TITLE
Question: Is there a way to use stylelint inside of gulp without using postcss?

### ISSUE THREAD

**Issue by josh18** [stylelint/stylelint#526]
I assume not as the only examples I've seen have used postcss but I thought I would ask just in case.

**Comment by TrySound** [comment]
@josh18 Stylelint in any case uses postcss. Does it too difficult just load gulp-postcss?

**Comment by josh18** [comment]
Well I my css is actually less and I don't think that works with postcss? Maybe my best option is to process the less file into css and then put the plain css through postcss for the stylelint output?

**Comment by TrySound** [comment]
@josh18 stylelint should process less source files. But maybe not all rules will do what you expect.

**Comment by TrySound** [comment]
@josh18 postcss is parser. Stylelint is just processor of ast

**Comment by davidtheclark** [comment]
@josh18 There is not a gulp-stylelint plugin at this time (you're welcome to make one), but you can either, as @TrySound said, use gulp-postcss, or you can use the standalone Node.js API within a gulp task.

**Comment by josh18** [comment]
I'll give it a shot, thanks for your help.

**Comment by josh18** [comment]
Ok so there doesn't look like there is any way to process LESS files. I assume there is no way to use sourcemaps so that stylelint outputs relevant filenames / lines?

**Comment by TrySound** [comment]
@josh18 Stylelint do not change your code. It's just linter. And you can process less files as I know. Or you get some error?

**Comment by josh18** [comment]
Yeah unfortunately [less mixins](http://lesscss.org/features/#features-overview-feature-mixins) give me the error "Unknown word". It doesn't seem to like the full stop syntax e.g. `.bordered;`.

**Comment by davidtheclark** [comment]
@josh18 Good reason to build (or push somebody to build) a less parser for PostCSS.

But also, you should be able to process your Less files (with Less) then feed the result *with the sourcemap* into PostCSS. You're using gulp? Why not just wrap both parts of the pipeline with gulp-sourcemaps? Does that not work?

**Comment by josh18** [comment]
Yeah a LESS parser would be ideal but unfortunately I think it is outside of my skill/time constraints at the moment.

Not quite sure what you mean but this is what I have so far:
```js
// styles.css = compiled css file
// styles.css.map = map from less compilation
return gulp.src('styles.css')
    .pipe(postcss([
        stylelint({
            rules: {
                indentation: [1, 'tab']
            }
        }, {
            map: {
                inline: false,
                prev:  'styles.css.map'
            }
        }),
        reporter({ clearMessages: true })
    ]));
```
Is this what you were intending? (Currently doesn't seem to work, stylelint just outputs line numbers from style.css rather than getting them from styles.css.map)

**Comment by davidtheclark** [comment]
Read the docs for both gulp-less and gulp-postcss about sourcemaps; that should give you the details. Something like this (untested code below):

```js
return gulp.src('styles.css')
  .pipe(gulpSourcemaps.init())
  .pipe(gulpLess())
  .pipe(gulpPostcss([..]));
```

**Comment by josh18** [comment]
Hmm I tried that but stylelint still doesn't seem to pick up the original file names / lines when processing.

Don't worry though I just realised that checking the style after the less compilation is useless anyways as less essentially rewrites all of your css so everything is changed from the original format. It looks like the only option is to have a less postcss parser.

Thanks for all your time and help!

**Comment by lkraav** [comment]
It looks like some movement is happening at https://github.com/gilt/postcss-less and https://github.com/webschik/postcss-less but nothing that could be considered stable.

**Comment by josh18** [comment]
Yeah I've been keeping an eye on those. Also haven't tried it yet but this might be a good solution for some people: https://github.com/lesshint/lesshint

**Comment by lkraav** [comment]
> https://github.com/lesshint/lesshint

Nice find. It's a bit sad to see so much complex functionality duplication across tools.
